src/ExtUvLoop.php

```diff
@@ -313,4 +313,32 @@ private function createStreamListener()
 
         return $callback;
     }
+
+    /**
+     * @param float $interval
+     * @param bool $allowZero
+     * @return int
+     */
+    private function convertFloatSecondsToMilliseconds($interval, $allowZero = false)
+    {
+        if ($interval < 0) {
+            $interval = 0;
```

Comment:
Might as well do a return here instead of going through all these calculations?

Reply:
It could return early only if i duplicate `$allowZero`-related logic from last line:
```php
return $result === 0 && !$allowZero
    ? 1
    : $result
;
```
So new code will look like:
```php
if ($interval < 0) {
    return $allowZero
        ? 0
        : 1
    ;
}
```
IMO it is better to leave it as is.

Reply:
I'm not sure for what it even is? Libuv allows 0, so differentiating between 0 and 1 is imo unnecessary.

Reply:
I must admit i don't know, but my guess is `addPeriodicTimer` with `0` will run code only once, also adding `1` was part of old code: https://github.com/reactphp/event-loop/blob/v1.1.0/src/ExtUvLoop.php#L130 and https://github.com/reactphp/event-loop/blob/v1.1.0/src/ExtUvLoop.php#L153-L154

Reply:
We only need a non-zero repeat (periodic timer). And then it'd be better to do an inline check there, so it only affects the repeat value and not the timeout value.

Reply:
Good catch, actually it makes conversion easier. Please check new code.